tensorflow-frameworks/tensorflow-data/src/test/java/org/tensorflow/data/BatchDatasetTest.java

```diff
@@ -0,0 +1,96 @@
+package org.tensorflow.data;
+
+import org.junit.Test;
+import org.tensorflow.Output;
+import org.tensorflow.Tensor;
+import org.tensorflow.op.Ops;
+import org.tensorflow.types.TInt32;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertTrue;
+
+public class BatchDatasetTest extends DatasetTestBase {
+
+  @Test
+  public void testEagerBatchDataset() {
+    Ops tf = Ops.create();
+
+    // EVEN BATCH SIZES
+    Dataset dataset = Dataset
+        .fromTensorSlices(tf,
+            Arrays.asList(
+                tf.val(testMatrix1),
+                tf.val(testMatrix2)),
+            Arrays.asList(TInt32.DTYPE, TInt32.DTYPE))
+        .batch(2);
+
+    int count = 0;
+
+    for (List<Output<?>> components : dataset) {
+      Tensor<TInt32> batch1 = components.get(0).tensor().expect(TInt32.DTYPE);
+      Tensor<TInt32> batch2 = components.get(1).tensor().expect(TInt32.DTYPE);
```

Comment:
As said before, tensors must be released with `try-with-resources` blocks

Reply:
Thanks, updated!